app/models/generic_announcement.rb

```diff
@@ -0,0 +1,8 @@
+# Represents a generic announcement, which may be either a system-level or instance-level one.
+class GenericAnnouncement
+  include ActiveModel::Model
+
+  def self.all
+    SystemAnnouncement.all + Instance::Announcement.all
```

Comment:
You might only want to define a method that returns only the active announcements. `all` can return quite a large number of records over time.